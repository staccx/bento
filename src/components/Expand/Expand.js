import React, { useState } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { BounceIn } from "../../animations"
import Caret from "../Icons/Caret/Caret"
import ThemeComponent from "../Theme/ThemeComponent/ThemeComponent"
import { applyVariants, color, commonPropTypes, spacing } from "../../theming"
import themeProps from "./Expand.themeProps"
import { componentCreateFactory } from "../../theming/utils/createVariantsFunctionFactory"

/**
 * Expand is a component that lets you simply create the posibility to have a heading with an onClick that displays
 an underlying object
 */
const Expand = ({
  className,
  expanded = false,
  closeOnItemClick,
  onClick,
  children,
  title,
  hideIcon,
  variant,
  id,
  ...restProps
}) => {
  const [isExpanded, setIsExpanded] = useState(expanded)

  const handleItemClick = e => {
    if (closeOnItemClick) {
      setIsExpanded(false)
    }
  }

  const handleExpand = event => {
    setIsExpanded(!isExpanded)

    if (onClick) {
      onClick(event)
    }
  }

  React.useEffect(() => {
    setIsExpanded(expanded)
  }, [expanded])

  return (
    <ExpandWrapper className={className} variant={variant} {...restProps}>
      {React.Children.map(children, (child, i) => {
        // Ignore the first child
        if (i < 1) {
          return (
            <ExpandBtn
              title={title}
              isExpanded={isExpanded}
              onClick={handleExpand}
              aria-expanded={isExpanded}
              aria-controls={title}
              id={(id ?? title) + "2"}
              variant={variant}
              type="button"
            >
              {child}
              {!hideIcon && (
                <ExpandIcon variant={variant} isExpanded={isExpanded} />
              )}
            </ExpandBtn>
          )
        }
        return (
          isExpanded && (
            <ExpandItem
              id={`item-${i}`}
              aria-labelledby={(id ?? title) + "2"}
              variant={variant}
              onClick={handleItemClick}
            >
              {child}
            </ExpandItem>
          )
        )
      })}
    </ExpandWrapper>
  )
}

const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={themeProps.iconComponent.name}
    fallback={Caret}
    {...props}
  />
)

const ExpandIcon = styled(IconComponent)`
  position: absolute;
  right: ${color.small};
  top: 50%;
  fill: ${color.wcag};
  transition: transform 0.3s ease-out;
  right: 0;
  transform: ${p =>
    p.isExpanded
      ? "translateY(-50%) rotate(180deg)"
      : "translateY(-50%) rotate(0)"};
  ${applyVariants(themeProps.icon.name)};
`

export const ExpandWrapper = styled.div`
  ${applyVariants(themeProps.wrapper.name)};
`

export const ExpandBtn = styled.button`
  position: relative;
  background-color: transparent;
  border-width: 0;
  color: inherit;
  padding: 0;
  text-align: left;
  padding-right: ${spacing.mediumPlus};
  margin: 0;
  min-height: 0;
  width: 100%;

  &:hover,
  &:focus,
  &:active {
    background-color: transparent;
    border-width: 0;
    color: inherit;
    outline: none;

    ${ExpandIcon} {
      fill: ${color.primary};
    }
  }

  ${applyVariants(themeProps.button.name)};
`

export const ExpandItem = styled.div`
  opacity: 0;
  animation: 0.4s ${BounceIn} 0.05s ease-out forwards 1;
  ${applyVariants(themeProps.item.name)};
`

Expand.defaultProps = {
  expanded: false,
  className: "",
  title: "",
  hideIcon: false,
  closeOnItemClick: false
}

Expand.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * Title element
   */
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array
  ]),
  className: PropTypes.string,
  /**
   * Boolean, set to true opens the Expand
   */
  expanded: PropTypes.bool,
  /**
   * Boolean, set to true hides the Icon
   */
  hideIcon: PropTypes.bool,
  /**
   * Set this to true if you want the expand to close after an element has been clicked
   */
  closeOnItemClick: PropTypes.bool
}
Expand.themeProps = themeProps
Expand.createVariants = componentCreateFactory(Expand)

export default Expand
