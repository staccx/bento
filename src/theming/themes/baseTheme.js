import React from "react"
import Theme from "../Theme"
import reset from "../styles/cssResets"
import grids from "../../components/Layout/Layout/Layout.Grids"
import codeTheme from "../codeTheme"

const Logo = () => <div>ThemeComponent for BaseTheme</div>
Logo.displayName = "Logo for base theme"

export const theme = {
  name: "Base Theme",
  storybookName: "Base Theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190", // for toggle eks
    link: "#279AF1"
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px",
    lineHeight: "1.5"
  },
  fontWeight: {
    bold: "bold",
    normal: "normal",
    light: "lighter"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px",
  reset,
  grid: grids,
  logo: Logo
}

export const StyleguideTheme = new Theme(theme, codeTheme)

export default theme
