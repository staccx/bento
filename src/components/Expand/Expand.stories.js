import React from "react"
import Expand from "./Expand"

export default {
  title: "Components/Expand",
  component: Expand
}

export const ExpandSpan = args => (
  <Expand title="click to see">
    <span>Click me to see</span>
    <div>
      <p>Some more text</p>
    </div>
  </Expand>
)

export const ExpandP = args => (
  <Expand>
    <p>Click me to see</p>
    <div>
      <p>Even more text</p>
    </div>
  </Expand>
)

export const ExpandDiv = args => (
  <Expand>
    <div>
      <p>All of me</p>
      <p>Can be clicked to show</p>
    </div>
    <div>This text is also possible to read</div>
  </Expand>
)

export const ExpandShowUl = args => (
  <Expand>
    <span>Expand me</span>
    <div>
      <ul>
        <li>I am hidden</li>
        <li> Until you expand</li>
      </ul>
    </div>
  </Expand>
)

export const ExpandNoHideIcon = args => (
  <Expand {...args}>
    <span>Click me to see</span>
    <div>
      <p>this</p>
    </div>
  </Expand>
)

ExpandNoHideIcon.args = { hideIcon: true }

export const ExpandExpanded = args => (
  <Expand {...args}>
    <span>Click me to hide</span>
    <div>
      <p>this</p>
    </div>
  </Expand>
)

ExpandExpanded.args = { expanded: true }
