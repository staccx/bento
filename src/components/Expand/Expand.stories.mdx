import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Expand from "./Expand.js"

<Meta title="Components|Expand" component={Expand} parameters={{component: Expand.themeProps.button.name}} />

# Expand

Expand is a component that lets you simply create the posibility to have a heading with an onClick that displays
an underlying object


```javascript
import {Expand} from "@staccx/bento"
```

## Expand with span as element
<Preview>
  <Story name="Expand span">
    <Expand title="click to see">
        <span>Click me to see</span>
            <div>
                <p>Some more text</p>
            </div>
    </Expand>
  </Story>
</Preview>

## Expand with p as element
<Preview>
  <Story name="Expand p">
    <Expand>
        <p>Click me to see</p>
            <div>
                <p>Even more text</p>
            </div>
    </Expand>
  </Story>
</Preview>

## Expand with div as element
<Preview>
  <Story name="Expand div">
    <Expand>
        <div>
           <p>All of me</p>
           <p>Can be clicked to show</p>
        </div>
            <div>
                This text is also possible to read
            </div>
    </Expand>
  </Story>
</Preview>

## Expand show ul
<Preview>
  <Story name="Expand show ul">
    <Expand>
      <span>Expand me</span>
      <div>
        <ul>
          <li>I am hidden</li>
          <li> Until you expand</li>
        </ul>
      </div>
    </Expand>
  </Story>
</Preview>

## Expand without hideIcon
The expand component has a prop called hideIcon. This allows you to remove or show the icon next to the text
on the Expand element. Default for hideIcon is set to false

<Preview>
  <Story name="Expand no hideIcon">
    <Expand hideIcon={true}>
        <span>Click me to see</span>
            <div>
                <p>this</p>
            </div>
    </Expand>
  </Story>
</Preview>

## Expand that is expanded as default
<Preview>
  <Story name="Expand expanded default">
    <Expand expanded={true}>
        <span>Click me to hide</span>
            <div>
                <p>this</p>
            </div>
    </Expand>
  </Story>
</Preview>

## Expand that close when auto when an item is clicked
<Preview>
  <Story name="Expand closeOnItemClick">
    <Expand closeOnItemClick>
        <span>Header item</span>
            <div>
                <p>Click any item to close</p>
            </div>
            <div>
                <p>Click any item to close</p>
            </div>
            <div>
                <p>Click any item to close</p>
            </div>
    </Expand>
  </Story>
</Preview>

### Source code

<details><summary>Expand</summary>
<p>

```bash
import React, { useState } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { BounceIn } from "../../animations"
import Caret from "../Icons/Caret/Caret"
import ThemeComponent from "../Theme/ThemeComponent"
import { applyVariants, color, spacing, commonPropTypes } from "../../theming"
import themeProps from "./Expand.themeProps"
import { componentCreateFactory } from "../../theming/utils/createVariantsFunctionFactory"

const Expand = ({
  className,
  expanded = false,
  onClick,
  children,
  title,
  hideIcon,
  variant,
  ...restProps
}) => {
  const [isExpanded, setIsExpanded] = useState(expanded)

  const handleExpand = event => {
    setIsExpanded(!isExpanded)

    if (onClick) {
      onClick(event)
    }
  }

  return (
    <ExpandWrapper className={className} variant={variant} {...restProps}>
      {React.Children.map(children, (child, i) => {
        // Ignore the first child
        if (i < 1) {
          return (
            <ExpandBtn
              title={title}
              isExpanded={isExpanded}
              onClick={e => handleExpand(e)}
              aria-expanded={isExpanded}
              aria-controls={title}
              id={title + "2"}
              variant={variant}
            >
              {child}
              {!hideIcon && (
                <ExpandIcon variant={variant} isExpanded={isExpanded} />
              )}
            </ExpandBtn>
          )
        }
        return (
          isExpanded && (
            <ExpandItem
              id={title}
              aria-labelledby={title + "2"}
              variant={variant}
            >
              {child}
            </ExpandItem>
          )
        )
      })}
    </ExpandWrapper>
  )
}

const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={themeProps.iconComponent.name}
    fallback={Caret}
    {...props}
  />
)

const ExpandIcon = styled(IconComponent)`
  position: absolute;
  right: ${color.small};
  top: 50%;
  fill: ${color.wcag};
  transition: transform 0.3s ease-out;
  right: 0;
  transform: ${p =>
    p.isExpanded
      ? "translateY(-50%) rotate(180deg)"
      : "translateY(-50%) rotate(0)"};
  ${applyVariants(themeProps.icon.name)};
`

export const ExpandWrapper = styled.div`
  ${applyVariants(themeProps.wrapper.name)};
`

export const ExpandBtn = styled.button`
  position: relative;
  background-color: transparent;
  border-width: 0;
  color: inherit;
  padding: 0;
  text-align: left;
  padding-right: ${spacing.mediumPlus};
  margin: 0;
  min-height: 0;
  width: 100%;

  &:hover,
  &:focus,
  &:active {
    background-color: transparent;
    border-width: 0;
    color: inherit;
    outline: none;

    ${ExpandIcon} {
      fill: ${color.primary};
    }
  }

  ${applyVariants(themeProps.button.name)};
`

export const ExpandItem = styled.div`
  opacity: 0;
  animation: 0.4s ${BounceIn} 0.05s ease-out forwards 1;
  ${applyVariants(themeProps.item.name)};
`

Expand.defaultProps = {
  expanded: false,
  className: "",
  title: "",
  hideIcon: false
}

Expand.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * Title element
   */
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array
  ]),
  className: PropTypes.string,
  /**
   * Boolean, set to true opens the Expand
   */
  expanded: PropTypes.bool,
  /**
   * Boolean, set to true hides the Icon
   */
  hideIcon: PropTypes.bool
}
Expand.themeProps = themeProps
Expand.createVariants = componentCreateFactory(Expand)

export default Expand

```

</p>
</details>



## Props for Expand

<Props of={Expand}/>
