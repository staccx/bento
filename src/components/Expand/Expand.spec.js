import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import { fireEvent, render, screen } from "@testing-library/react"
import baseTheme from "../../theming/themes/baseTheme"
import Expand from "./Expand"

const SpanContent = () => <p>header</p>
const Child = ({ text = "Body" }) => <p>{text}</p>

describe("Expand", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand>
              <span>
                <Child />
              </span>
              <Child />
            </Expand>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("With title", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand title="Title">
              <span>
                <Child />
              </span>
              <Child />
            </Expand>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("With console", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand title="Title" onClick={e => console.log("console")}>
              <span>
                <Child />
              </span>
              <Child />
            </Expand>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("No children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Expanded true", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand expanded>
              <span>
                <Child />
              </span>
              <Child />
            </Expand>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid Expand", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand expanded="true">
              <span>
                <Child />
              </span>
              <Child />
            </Expand>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("hide icon", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Expand hideIcon>
              <span>
                <Child />
              </span>
              <Child />
            </Expand>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
  describe("Rendering", () => {
    it("should show child onClick", () => {
      render(
        <ThemeProvider theme={baseTheme}>
          <Expand>
            <span>
              <SpanContent />
            </span>
            <Child />
          </Expand>
        </ThemeProvider>
      )
      expect(() => screen.getByText("Body")).toThrow()
      fireEvent.click(screen.getByText("header"))
      expect(screen.getByText("Body")).toBeInTheDocument()
    })
    it("pass onClick", () => {
      render(
        <ThemeProvider theme={baseTheme}>
          <Expand
            onClick={() => {
              console.log("onClick")
            }}
          >
            <span>
              <SpanContent />
            </span>
            <Child />
          </Expand>
        </ThemeProvider>
      )
      expect(() => screen.getByText("Body")).toThrow()
      fireEvent.click(screen.getByText("header"))
      expect(screen.getByText("Body")).toBeInTheDocument()
    })
    it("Expanded false", () => {
      render(
        <ThemeProvider theme={baseTheme}>
          <Expand expanded={false}>
            <span>
              <SpanContent />
            </span>
            <Child />
          </Expand>
        </ThemeProvider>
      )
      expect(() => screen.getByText("Body")).toThrow()
      fireEvent.click(screen.getByText("header"))
      expect(screen.getByText("Body")).toBeInTheDocument()
    })

    it("Should close on item click if closeOnItemClick is set", () => {
      render(
        <ThemeProvider theme={baseTheme}>
          <Expand expanded closeOnItemClick>
            <span>
              <SpanContent />
            </span>
            <Child />
            <Child text="Second child" />
          </Expand>
        </ThemeProvider>
      )
      expect(screen.getByText("Body")).toBeInTheDocument()
      fireEvent.click(screen.getByText("Second child"))
      expect(() => screen.getByText("Body")).toThrow()
    })
  })
})
