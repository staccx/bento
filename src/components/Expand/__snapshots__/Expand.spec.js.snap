// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expand Snapshots Expanded true 1`] = `
<div
  className="sc-AxirZ feQoIa"
>
  <button
    aria-controls=""
    aria-expanded={true}
    className="sc-AxiKw eAKUnW"
    id="2"
    onClick={[Function]}
    title=""
  >
    <span>
      <p>
        Body
      </p>
    </span>
    <svg
      className="sc-AxjAm icchVr"
      height="7"
      viewBox="0 0 33 20"
      width="12"
    >
      <path
        d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
      />
    </svg>
  </button>
  <div
    aria-labelledby="2"
    className="sc-AxhCb fnwEPP"
    id="item-1"
    onClick={[Function]}
  >
    <p>
      Body
    </p>
  </div>
</div>
`;

exports[`Expand Snapshots No children 1`] = `
<div
  className="sc-AxirZ feQoIa"
/>
`;

exports[`Expand Snapshots No props 1`] = `
<div
  className="sc-AxirZ feQoIa"
>
  <button
    aria-controls=""
    aria-expanded={false}
    className="sc-AxiKw eAKUnW"
    id="2"
    onClick={[Function]}
    title=""
  >
    <span>
      <p>
        Body
      </p>
    </span>
    <svg
      className="sc-AxjAm docPTU"
      height="7"
      viewBox="0 0 33 20"
      width="12"
    >
      <path
        d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
      />
    </svg>
  </button>
</div>
`;

exports[`Expand Snapshots With console 1`] = `
<div
  className="sc-AxirZ feQoIa"
>
  <button
    aria-controls="Title"
    aria-expanded={false}
    className="sc-AxiKw eAKUnW"
    id="Title2"
    onClick={[Function]}
    title="Title"
  >
    <span>
      <p>
        Body
      </p>
    </span>
    <svg
      className="sc-AxjAm docPTU"
      height="7"
      viewBox="0 0 33 20"
      width="12"
    >
      <path
        d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
      />
    </svg>
  </button>
</div>
`;

exports[`Expand Snapshots With title 1`] = `
<div
  className="sc-AxirZ feQoIa"
>
  <button
    aria-controls="Title"
    aria-expanded={false}
    className="sc-AxiKw eAKUnW"
    id="Title2"
    onClick={[Function]}
    title="Title"
  >
    <span>
      <p>
        Body
      </p>
    </span>
    <svg
      className="sc-AxjAm docPTU"
      height="7"
      viewBox="0 0 33 20"
      width="12"
    >
      <path
        d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
      />
    </svg>
  </button>
</div>
`;

exports[`Expand Snapshots hide icon 1`] = `
<div
  className="sc-AxirZ feQoIa"
>
  <button
    aria-controls=""
    aria-expanded={false}
    className="sc-AxiKw eAKUnW"
    id="2"
    onClick={[Function]}
    title=""
  >
    <span>
      <p>
        Body
      </p>
    </span>
  </button>
</div>
`;

exports[`Expand Snapshots invalid Expand 1`] = `
<div
  className="sc-AxirZ feQoIa"
>
  <button
    aria-controls=""
    aria-expanded="true"
    className="sc-AxiKw eAKUnW"
    id="2"
    onClick={[Function]}
    title=""
  >
    <span>
      <p>
        Body
      </p>
    </span>
    <svg
      className="sc-AxjAm icchVr"
      height="7"
      viewBox="0 0 33 20"
      width="12"
    >
      <path
        d="M16.5 12.15L28.19.87c.52-.5 1.35-.5 1.87.01l2.05 2.02a1.3 1.3 0 0 1 0 1.86L17.44 19.12a1.36 1.36 0 0 1-1.88 0L.9 4.76A1.3 1.3 0 0 1 .89 2.9L2.94.88c.52-.5 1.35-.5 1.87 0L16.5 12.14v.01z"
      />
    </svg>
  </button>
  <div
    aria-labelledby="2"
    className="sc-AxhCb fnwEPP"
    id="item-1"
    onClick={[Function]}
  >
    <p>
      Body
    </p>
  </div>
</div>
`;
